### ISSUE TITLE
`borsh` related compilation issue due to `mpl-token-auth-rules` bump

### ISSUE THREAD

**Issue by acheroncrypto** [metaplex-foundation/mpl-token-metadata#28]
### Problem

`mpl-token-metadata` doesn't compile due to the latest release of `mpl-token-auth-rules` [bumping `borsh` version to `0.10.3`](https://github.com/metaplex-foundation/mpl-token-auth-rules/pull/140/files#diff-3990cb83308a149fa4ca6cf5b0d8eab249c9ba60fba350fcccc29f00de837bdfR13).

```toml
borsh = "0.10.3"
solana-program = ">= 1.14.13, < 1.17"
```

Allowing different versions like this won't help because `solana-program 1.14` doesn't work with `borsh 0.10.3`. The solution would be either supporting both `borsh` versions like [this](https://github.com/coral-xyz/anchor/blob/b7b8736276ca711e76199767153ac24c4a60d9f9/lang/Cargo.toml#L54) or drop the support for `solana-program 1.14`.

```toml
[dependencies]
mpl-token-metadata = "1.13.1"
```

This is enough to cause `borsh` related compile issues because it will pull the latest version of `mpl-token-auth-rules 1.4.2`.

Note that this is not related to this specific version of `mpl-token-metadata`, same problem will happen on any previous version that have dependency `mpl-token-auth-rules ^1`.

### Solution

Short term solution for people who run into this would be to downgrade `mpl-token-auth-rules` by running:

```
cargo update -p mpl-token-auth-rules --precise 1.4.1
```

Long term, it would help to decrease dependency issues like this if `mpl` programs followed the same dependency versions, especially when there is a breaking change between versions(whether external or not).

**Comment by febo** [comment]
This has been addressed in the latest Token Auth Rules crate `1.4.3`. The solution was to allow a range on the borsh dependency.

This is the PR with the change: https://github.com/metaplex-foundation/mpl-token-auth-rules/pull/141

**Comment by febo** [comment]
Investigating this further.

**Comment by febo** [comment]
Got to the bottom of it.

@acheroncrypto The build issue was fixed by the Token Auth Rules `1.4.3`, which introduced the range for the borsh dependency.

The issue with Rooster was only observed when running bpf tests. The solution was to pin the dependency to a specific commit revision (here is the PR: https://github.com/metaplex-foundation/mpl-token-metadata/pull/30).
